From: SeongJae Park <sj@kernel.org>
Date: Tue, 2 Dec 2025 22:32:57 -0800
Subject: [PATCH] mm/damon/core: report whether the page fault was for writing

The page fault event reporting function can know if the access was for
writing or reading.  Add the information to the report.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 2288c541d7a8a..7087a184c00ce 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1801,6 +1801,7 @@ void damon_report_page_fault(struct vm_fault *vmf, bool huge_pmd)
 		.size = 1,	/* todo: set appripriately */
 		.cpu = smp_processor_id(),
 		.tid = task_pid_vnr(current),
+		.is_write = vmf->flags & FAULT_FLAG_WRITE,
 	};
 
 	if (huge_pmd)
-- 
2.47.3

